File: iterator/python/employee_list_aggregate.py
from employee_aggregate import EmployeeAggregate
from employee_iterator import EmployeeIterator
from employee import Employee


class EmployeListAggregate(EmployeeAggregate):
    """
    Aggregate (IEmployeeAggregate) arayüzünü uygular ve Iterator arayüzünü uygulayan sınıf örneği üretir.
    Değerlerin referansları burada bulunur.
    UML diyagramındaki ConcreteAggregate  sınıfına denk gelmektedir.
    """

    def __init__(self):
        self._employees = []

    def add(self, employee: Employee):
        self._employees.append(employee)

    def count(self):
        return len(self._employees)

    def get(self, index: int):
        return self._employees[index]

    def create_iterator(self):
        return EmployeeIterator(self)

File: iterator/typescript/employee-list-aggregate.ts
import {IEmployeeAggregate} from "./employee-aggregate.interface";
import {Employee} from "./employee";
import {IIterator} from "./iterator.interface";
import {EmployeeIterator} from "./employee-iterator";

// Aggregate (IEmployeeAggregate) arayüzünü uygular ve Iterator arayüzünü uygulayan sınıf örneği üretir.
// Değerlerin referansları burada bulunur.
// UML diyagramındaki ConcreteAggregate  sınıfına denk gelmektedir.
export class EmployeeListAggregate implements IEmployeeAggregate {
  private readonly employees: Employee[] = [];

  public add(employee: Employee): void {
    this.employees.push(employee);
  }

  public count(): number {
    return this.employees.length;
  }

  public get(index: number): Employee {
    return this.employees[index];
  }

  public createIterator(): IIterator<Employee> {
    return new EmployeeIterator(this);
  }
}
